src/mlpack/methods/ann/layer/group_norm.hpp

```diff
@@ -0,0 +1,204 @@
+{
+ public:
+  //! Create the GroupNorm object.
+  GroupNorm();
+
+  /**
+   * Create the GroupNorm object for a specified number of input units.
+   *
+   * @param size The number of input units.
+   * @param eps The epsilon added to variance to ensure numerical stability.
+   */
+  GroupNorm(const size_t size, const size_t groupCount, const double eps = 1e-8);
+
+  /**
+   * Reset the layer parameters.
+   */
+  void Reset();
+
+  /**
+   * Forward pass of Group Normalization. Transforms the input data
+   * into zero mean and unit variance, scales the data by a factor gamma and
+   * shifts it by beta.
+   *
+   * @param input Input data for the layer.
+   * @param output Resulting output activations.
+   */
+  template<typename eT>
+  void Forward(const arma::Mat<eT>& input, arma::Mat<eT>& output);
+
+  /**
+   * Backward pass through the layer.
+   *
+   * @param input The input activations.
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename eT>
+  void Backward(const arma::Mat<eT>& input,
+                const arma::Mat<eT>& gy,
+                arma::Mat<eT>& g);
+
+  /**
+   * Calculate the gradient using the output delta and the input activations.
+   *
+   * @param input The input activations.
+   * @param error The calculated error.
+   * @param gradient The calculated gradient.
+   */
+  template<typename eT>
+  void Gradient(const arma::Mat<eT>& input,
+                const arma::Mat<eT>& error,
+                arma::Mat<eT>& gradient);
+
+  //! Get the parameters.
+  OutputDataType const& Parameters() const { return weights; }
+  //! Modify the parameters.
+  OutputDataType& Parameters() { return weights; }
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the gradient.
+  OutputDataType const& Gradient() const { return gradient; }
+  //! Modify the gradient.
+  OutputDataType& Gradient() { return gradient; }
+
+  //! Get the mean across single training data.
+  OutputDataType Mean() { return mean; }
+
+  //! Get the variance across single training data.
+  OutputDataType Variance() { return variance; }
+
+  //! Get the number of input units.
+  size_t InSize() const { return size; }
+
+  //! Get the value of epsilon.
+  double Epsilon() const { return eps; }
+
+  //! Get the shape of the input.
+  size_t InputShape() const
+  {
+    return size;
+  }
+
+  //! Get the group count.
+  size_t GroupCount() const
+  {
+    return groupCount;
+  }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const uint32_t /* version */);
+
+ private:
+  //! Locally-stored number of input units.
+  size_t size;
```

Comment:
```suggestion
  //! Locally-stored number of channels.
  size_t channelSize;
  
  //! Locally-stored number of input units.
  size_t size;
```

To fix the build issue.